### ISSUE TITLE
LibWeb: Hidden overflow is not hidden on X.com explore page

### ISSUE THREAD

**Issue by awesomekling** [SerenityOS/serenity#22883]
Example:

![image](https://github.com/SerenityOS/serenity/assets/5954907/c3818ec6-6988-458b-ac75-35b46b667605)

Simplified reduction:

```html
<!doctype html><style>
  body {
    width: 100px;
    height: 100px;
  }
  #outer {
    width: 200px;
    height: 100px;
    overflow: hidden;
  }
  #inner {
    position: relative;
    z-index: 0;
    overflow: hidden;
  }
  #green {
    width: 200px;
    height: 200px;
    background: green;
  }
</style><body><div id="outer"><div id="inner"><div id="green">
```

**Comment by kalenikaliaksandr** [comment]
another example that we don't render correctly right now but could be used as a test in the future (I hope):
```html
<style>
    .box {
        width: 100px;
        height: 100px;
        background-color: magenta;
        opacity: 0.5;
        overflow: hidden;
    }

    .relpos {
        position: relative;
    }

    .abspos {
        position: absolute;
        top: 150px;
        left: 150px;
        width: 50px;
        height: 50px;
        background-color: yellow;
        font-size: 50px;
    }
</style>
<div class="relpos">
<div class="box">
    <div class="abspos"></div>
</div>
</div>
```

currently we crop yellow div by overflow of magenta div although it is not contained by magenta div.

**Comment by AtkinsSJ** [comment]
Fixed by #22970.
